core/src/Kokkos_CopyViews.hpp

```diff
@@ -1043,9 +1059,9 @@ inline void deep_copy(
   using dst_memory_space = typename dst_type::memory_space;
   using src_memory_space = typename src_type::memory_space;
 
-  static_assert(std::is_same_v<typename dst_type::value_type,
-                               typename src_type::non_const_value_type>,
-                "deep_copy requires matching non-const destination type");
+  static_assert(
+      std::is_same_v<value_type, typename src_type::non_const_value_type>,
+      "deep_copy requires matching non-const destination type");
```

Comment:
ok, sure

Reply:
Yeah this may seem a driveby fix but if `KOKKOS_ENABLE_IMPL_VIEW_LEGACY` is not set `value_type` is not used. I could have marked it as `[[maybe_unused]]` or only defined it when using legacy view but I saw that this line could be using the alias but wasn't. It solves the unused alias problem while being better code in general :P

Reply:
Sounds like a good solution.